### ISSUE TITLE
Wrong PHP include_path causes PHP_CodeSniffer to fail in Trusty

### ISSUE THREAD

**Issue by LogMANOriginal** [travis-ci/travis-ci#8487]
Our build script fails for all environments (except hhvm and nightly) when running PHP_CodeSniffer using the Ubuntu Trusty environment. The build succeeds when using the `dist: precise` option.

[failing build (Trusty)](https://travis-ci.org/RSS-Bridge/rss-bridge/builds/277092603)
[passing build (Precise)](https://travis-ci.org/RSS-Bridge/rss-bridge/builds/278986129)

PHP_CodeSniffer fails because it cannot find necessary files in the defined `include_path`

This bug can be reproduced by using following [script](https://github.com/RSS-Bridge/rss-bridge/blob/1ea091f2151cddb3410d94d72015f1fe5f1c53b9/.travis.yml) (requires a [phpcs.xml](https://github.com/RSS-Bridge/rss-bridge/blob/1ea091f2151cddb3410d94d72015f1fe5f1c53b9/phpcs.xml)):

```yaml
dist: trusty
language: php
php:
  - '5.6'
  - '7.0'
  - hhvm
  - nightly

install:
  - pear channel-update pear.php.net
  - pear install PHP_CodeSniffer

script:
  - phpenv rehash
  - phpcs . --standard=phpcs.xml --warning-severity=0 --extensions=php -p

matrix:
  fast_finish: true
  allow_failures:
    - php: hhvm
    - php: nightly
```

The build will fail for all environments except nightly (and hhvm, which is failing for different reasons). This is the failing [output](https://travis-ci.org/RSS-Bridge/rss-bridge/jobs/277092604)

```
$ phpcs . --standard=phpcs.xml --warning-severity=0 --extensions=php -p

PHP Warning:  include_once(PHP/CodeSniffer/autoload.php): failed to open stream: No such file or directory in /home/travis/.phpenv/versions/5.6.31/bin/phpcs on line 14

PHP Stack trace:

PHP   1. {main}() /home/travis/.phpenv/versions/5.6.31/bin/phpcs:0

Warning: include_once(PHP/CodeSniffer/autoload.php): failed to open stream: No such file or directory in /home/travis/.phpenv/versions/5.6.31/bin/phpcs on line 14

Call Stack:

    0.0001     240840   1. {main}() /home/travis/.phpenv/versions/5.6.31/bin/phpcs:0

PHP Warning:  include_once(): Failed opening 'PHP/CodeSniffer/autoload.php' for inclusion (include_path='.:/home/travis/.phpenv/versions/5.6.31/share/pear') in /home/travis/.phpenv/versions/5.6.31/bin/phpcs on line 14

PHP Stack trace:

PHP   1. {main}() /home/travis/.phpenv/versions/5.6.31/bin/phpcs:0

Warning: include_once(): Failed opening 'PHP/CodeSniffer/autoload.php' for inclusion (include_path='.:/home/travis/.phpenv/versions/5.6.31/share/pear') in /home/travis/.phpenv/versions/5.6.31/bin/phpcs on line 14

Call Stack:

    0.0001     240840   1. {main}() /home/travis/.phpenv/versions/5.6.31/bin/phpcs:0

PHP Fatal error:  Class 'PHP_CodeSniffer\Runner' not found in /home/travis/.phpenv/versions/5.6.31/bin/phpcs on line 17

PHP Stack trace:

PHP   1. {main}() /home/travis/.phpenv/versions/5.6.31/bin/phpcs:0

Fatal error: Class 'PHP_CodeSniffer\Runner' not found in /home/travis/.phpenv/versions/5.6.31/bin/phpcs on line 17

Call Stack:

    0.0001     240840   1. {main}() /home/travis/.phpenv/versions/5.6.31/bin/phpcs:0

The command "phpcs . --standard=phpcs.xml --warning-severity=0 --extensions=php -p" exited with 255.
```

The root cause for this issue is a wrong `include_path`, which is defined as (example from PHP 5.6)
`.:/home/travis/.phpenv/versions/5.6.31/share/pear` instead of
`.:/home/travis/.phpenv/versions/5.6.31/lib/php/pear`

This hotfix replaces the `include_path` during `before_install`  (not pretty, but it works)

```
before_install:
  - if [[ ${TRAVIS_PHP_VERSION:0:1} == "5" || ${TRAVIS_PHP_VERSION:0:1} == "7" ]]; then echo "include_path='.:/home/travis/.phpenv/versions/$(phpenv version-name)/lib/php/pear'" >> ~/.phpenv/versions/$(phpenv version-name)/etc/php.ini; fi
```

Hope that helps

**Comment by BanzaiMan** [comment]
I think this is a variant of #8365, in which `~/.pearrc` with incorrect content is in use. Try forcing a run-time installation; e.g., `php: 5.6.30` or `php: 7.0.23`.

**Comment by LogMANOriginal** [comment]
You are right, forcing versions other than the pre-installed ones fixes the problem: https://travis-ci.org/LogMANOriginal/rss-bridge/builds/279194153

It looks to me like the folder layout changed since Precise (`../share/` => `../lib/php/`), but the configuration was not updated accordingly. 

Not sure if this is related or not, but here is a bug report on the PEAR bug tracker ([Bug #17045 | pearcmd.php ignores / overrides include_path](https://pear.php.net/bugs/bug.php?id=17045)) that was fixed in [1.10.0](https://pear.php.net/package/PEAR/download/1.10.0)

**Comment by BanzaiMan** [comment]
OK. I'm closing this as a duplicate, then. I'll take a look soon.
